crates/astria-sequencer/src/bridge/state_ext.rs

```diff
@@ -265,60 +266,26 @@ pub(crate) trait StateReadExt: StateRead + address::StateReadExt {
-        let bytes = self
-            .nonverifiable_get_raw(&deposit_nonce_storage_key(rollup_id))
-            .await
-            .map_err(anyhow_to_eyre)
-            .wrap_err("failed reading raw deposit nonce from state")?;
-        let Some(bytes) = bytes else {
-            // no deposits for this rollup id yet; return 0
-            return Ok(0);
-        };
-
-        let Nonce(nonce) =
-            Nonce(u32::from_be_bytes(bytes.try_into().expect(
-                "all deposit nonces stored should be 4 bytes; this is a bug",
-            )));
-        Ok(nonce)
-    }
-
     #[instrument(skip_all)]
     async fn get_deposit_rollup_ids(&self) -> Result<HashSet<RollupId>> {
-        let mut stream = std::pin::pin!(self.nonverifiable_prefix_raw(DEPOSIT_PREFIX.as_bytes()));
+        let mut stream = std::pin::pin!(self.nonverifiable_prefix_raw(DEPOSIT_PREFIX));
         let mut rollup_ids = HashSet::new();
         while let Some(Ok((key, _))) = stream.next().await {
-            // the deposit key is of the form "deposit/{rollup_id}/{nonce}"
-            let key_str =
-                String::from_utf8(key).wrap_err("failed to convert deposit key to string")?;
-            let key_parts = key_str.split('/').collect::<Vec<_>>();
-            if key_parts.len() != 3 {
+            // the deposit key is of the form b"deposit/{32 bytes of rollup_id}{4 bytes of nonce}"
+            let Some(rollup_id_and_nonce) = key.strip_prefix(DEPOSIT_PREFIX) else {
+                error!("got wrong prefix in prefix stream");
                 continue;
-            }
-            let rollup_id_bytes =
-                hex::decode(key_parts[1]).wrap_err("invalid rollup ID hex string")?;
+            };
+            let Some(rollup_id_slice) = rollup_id_and_nonce.get(0..32) else {
+                bail!("failed to get rollup id bytes from deposit key {:?}", key);
+            };
             let rollup_id =
-                RollupId::try_from_slice(&rollup_id_bytes).wrap_err("invalid rollup ID bytes")?;
+                RollupId::try_from_slice(rollup_id_slice).wrap_err("invalid rollup ID bytes")?;
             rollup_ids.insert(rollup_id);
         }
         Ok(rollup_ids)
     }
 
-    #[instrument(skip_all)]
-    async fn get_deposit_events(&self, rollup_id: &RollupId) -> Result<Vec<Deposit>> {
-        let mut stream = std::pin::pin!(
-            self.nonverifiable_prefix_raw(deposit_storage_key_prefix(rollup_id).as_bytes())
-        );
-        let mut deposits = Vec::new();
-        while let Some(Ok((_, value))) = stream.next().await {
-            let raw = RawDeposit::decode(value.as_ref()).wrap_err("invalid deposit bytes")?;
-            let deposit = Deposit::try_from_raw(raw).wrap_err("invalid deposit raw proto")?;
-            deposits.push(deposit);
-        }
-        Ok(deposits)
-    }
-
     #[instrument(skip_all)]
     async fn get_block_deposits(&self) -> Result<HashMap<RollupId, Vec<Deposit>>> {
```

Comment:
Same here. This should be ideally a stream that's consumed into a datastructure at the call site.

Probaby best done in a followup.

Reply:
As above.